### ISSUE TITLE
Milestone-2: Basic Functionality of the Project

### ISSUE THREAD

**Issue by xltvy** [bounswe/bounswe2022group2#257]
### Issue Description

* As part of the Milestone Report 2 we need to provide a description of the basic functionality of the project under the section of the Executive Summary.
* I took the responsibility of putting **Basic Functionality of the Project** part together.
* **Basic Functionality of the Project** part will depict and summarize the functionalities of our practice application.

### Step Details

Steps that will be performed:
- [ ] Examine our application's backend and frontend.
- [ ] Write down a description of the basic functionality of the project
- [ ] Add the texts to appropriate parts of the online document

### Final Actions

* After above mentioned steps are done, this issue will be reviewed by the reviewer specified below before finalizing.
* Reviewer will review the issue and give feedbacks/suggestions if necessary.
* After necessary revisions are made, this issue will be closed.

### Deadline of the Issue

20.05.2022 - Friday - 17.00

### Reviewer

Ecenur Sezer

### Deadline for the Review

20.05.2022 - Friday - 18.00

**Comment by xltvy** [comment]
I have completed the basic functionalities of the project part of the milestone report 2, which can be seen from [this online document](https://docs.google.com/document/d/1g54mQYJxYoSeIKYAIz8FMHkJQ0-yczkttu_ddg0-PPs/edit). I am also providing the copied text below:

Our project’s aim was establishing a web application that has functionalities that eclipses with the work we have done until our previous milestone report 1. Since we have aimed to develop an online learning platform according to the project description provided, our practice application is developed to have some of the basic functionalities of the online learning platform we have planned throughout the semester.

Our practice application as mentioned above is a web application and provides a graphical user interface for users. A signup page welcomes users and they can sign up to our database by filling the required fields. Each user has a unique email and a hashed password stored in our database. A signed up user can log in to our application via the login page.

Since our practice application is some sort of a learning platform, there are three main components included alongside the user: lessons, categories, and events. Each user can create a lesson by providing required fields and then search for lessons available throughout the application. Each lesson belongs to a specific category, which can also be added to our application’s database by the users. To add a category, providing a valid category name is enough. Our application connects to Wikipedia’s public API to fetch category description and displays each category alongside its name in the categories page. Users can also filter the categories according to their names. When its category is added in our application, a user can create a lesson. Since each lesson has a category and a lecturer, which is the user who created it, a user can filter lessons by category and lecturers. According to their preferences a user can enroll to a lesson and rate it in the rating page. Rating page also provides users’ the opportunity to filter lessons according to their ratings.

For each lesson, a user can create an event that will occur in a specific location and on a specific date, by providing those fields alongside the title and the lesson ID of the lesson that the event belongs to. Each user can create an event for an existing lesson and upon creation our system assigns them as the host of the event. During creation, the user needs to provide a valid location and our application converts it to a valid full address via an API call to the Google Maps Geocoding API. Each user can view a lesson’s events alongside its specific event details that are specified during the creation and attend to an event. After a user attends an event, they can see the events they are attending in the attended events page.

To sum up, our application is a basic online learning platform that contains lessons under the section of categories and enables users to create and enroll to a lesson alongside creating and attending to an event about the lesson.

**Comment by codingAku** [comment]
I have read the functionality part. As far as I see, no feature was missed. Thank you.
